cmd/monaco/deploy/deploy.go

```diff
@@ -111,6 +121,64 @@ func deployConfigs(fs afero.Fs, manifestPath string, environmentGroups []string,
 	return nil
 }
 
+func checkUniquenessOfNameForClassicConfig(projects []project.Project) []error {
+	var errs []error
+	type (
+		environmentName = string
+		classicEndpoint = string
+	)
+	uniqueList := make(map[environmentName]map[classicEndpoint][]config.Config)
+	e := api.NewAPIs()
+
+	checkUniquenessOfName := func(c config.Config) {
+		a, ok := c.Type.(config.ClassicApiType)
+		if !ok || e[a.Api].NonUniqueName {
+			return
+		}
+
+		if uniqueList[c.Environment] == nil {
+			uniqueList[c.Environment] = make(map[classicEndpoint][]config.Config)
+		}
+
+		for _, c2 := range uniqueList[c.Environment][a.Api] {
+			n1, err := getNameForConfig(c)
+			if err != nil {
+				errs = append(errs, err)
+				return
```

Comment:
if I'm not mistaken this method returns whenever an error occurs. Does it really need a slice of errors of can you just return a single error/nil